WMF Framework/Event Platform/StorageManager.swift

```diff
@@ -0,0 +1,185 @@
+import Foundation
+import CocoaLumberjackSwift
+
+class StorageManager: NSObject {
+
+    private var managedObjectContext: NSManagedObjectContext
+    private let pruningAge: TimeInterval = 60*60*24*30 // 30 days
+
+    static let shared: StorageManager? = {
+        let fileManager = FileManager.default
+        var storageDirectory = fileManager.wmf_containerURL().appendingPathComponent("Event Platform", isDirectory: true)
+
+        do {
+            try fileManager.createDirectory(at: storageDirectory, withIntermediateDirectories: true, attributes: nil)
+        } catch {
+            do {
+                let storageDirectoryIsReachable = try storageDirectory.checkResourceIsReachable()
+                if !storageDirectoryIsReachable {
+                    DDLogError("EPC: Event Platform event storage directory is unreachable")
+                    return nil
+                }
+            } catch {
+                DDLogError("EPC: Error when attempting to reach Event Platform event storage directory")
+                return nil
+            }
+        }
+
+        do {
+            var values = URLResourceValues()
+            values.isExcludedFromBackup = true
+            try storageDirectory.setResourceValues(values)
+        } catch let error {
+            DDLogError("EPC: Error excluding from backup: \(error)")
```

Comment:
1. What kind of thing would cause setResourceValues to throw?
2. Should we be treating this as a StorageManager instantiation failure?

Reply:
The docs don't specify exactly what would cause `setResourceValues` to throw (though they specify things that don't 🙃 ).  At any rate, it does seems cleaner to me to keep all of this in a single block devoted to StorageManager instantiation.  I'll update accordingly.

Reply:
https://developer.apple.com/documentation/foundation/nsurl/1413819-setresourcevalue